packages/aws-cdk-lib/core/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable brace-style */
+import { Construct, IConstruct } from 'constructs';
+import { IAspect, Aspects } from './aspect';
+import * as cdk from '../../../aws-cdk-lib';
+import * as lambda from '../../aws-lambda';
+import { ReceiptRuleSet } from '../../aws-ses/lib/receipt-rule-set';
+import { Provider } from '../../custom-resources/lib';
+
+/**
+ * Runtime aspec
+ */
+abstract class RuntimeAspectsBase implements IAspect {
+
+  /**
+   * The string key for the runtime
+   */
+  public readonly targetRuntime: string;
+
+  constructor(runtime: string) {
+    this.targetRuntime = runtime;
+  }
+
+  public visit(construct: IConstruct): void {
+
+    //To handle custom providers
+    if (construct instanceof cdk.CustomResourceProviderBase) {
+      this.handleCustomResourceProvider(construct);
+    }
+
+    //To handle providers
+    else if (construct instanceof Provider) {
+      this.handleProvider(construct);
+    }
+
+    //To handle single Function case
+    else if (construct instanceof ReceiptRuleSet) {
+      this.handleReceiptRuleSet(construct);
+    }
+  }
+
+  private handleCustomResourceProvider(provider: cdk.CustomResourceProviderBase): void {
+    const node = provider as cdk.CustomResourceProviderBase;
+    // only replace for nodejs case
+    if (this.isValidRuntime(node.runtime)) {
+      const targetnode = node.node.children.find((child) => child instanceof cdk.CfnResource &&
+      child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      targetnode.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  private handleProvider(provider: Provider): void {
+    const functionNodes = provider.node.children.filter((child) => child instanceof lambda.Function);
+    for ( var node of functionNodes) {
+      const targetNode = node.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        targetNode.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      // onEvent Handlers
+      const onEventHandler = provider.onEventHandler as lambda.Function;
+      const onEventHandlerRuntime = onEventHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
```

Comment:
Rather than checking for `instanceof CfnResource` please use the `CfnResource.isResource()` static method.